Build failure on Windows due to conflicting slatedb.pdb output names

**Describe the bug**
After upgrading to radix_trie 0.3.0 which fixes #895, I hit an issue seemingly caused by slatedb-cli and slatedb-py having the same output filename (slatedb):

```
warning: output filename collision.
The lib target `slatedb` in package `slatedb-py v0.8.2 (C:\dev\slatedb\slatedb-py)` has the same output filename as the bin target `slatedb` in package `slatedb-cli v0.8.2 (C:\dev\slatedb\slatedb-cli)`.
Colliding filename is: C:\dev\slatedb\target\debug\deps\slatedb.pdb
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
```

This causes the build to fail at link time:

<details>

```
error: linking with `link.exe` failed: exit code: 1201
  |
  = note: "C:\\Program Files\\Microsoft Visual Studio\\18\\Preview\\VC\\Tools\\MSVC\\14.50.35503\\bin\\HostX64\\x64\\link.exe" "/DEF:C:\\Users\\rebond\\AppData\\Local\\Temp\\rustcG1lGzF\\lib.def" "/NOLOGO" "C:\\Users\\rebond\\AppData\\Local\\Temp\\rustcG1lGzF\\symbols.o" "<257 object files omitted>" "C:\\dev\\slatedb\\target\\debug\\deps/{libpyo3_async_runtimes-b76a9695cc13f6d9.rlib,libpyo3-6099d6c2ce5d339a.rlib,libpyo3_ffi-191588c5242bef9c.rlib,libunindent-ca4e93abb74204b8.rlib,libslatedb-95c70ec7d6a853ce.rlib,liblz4_flex-76f8b35e02f5ec07.rlib,libflate2-b1a251b89f4e7723.rlib,libminiz_oxide-6e0066e1564fab78.rlib,libadler2-169b89118f2b28c6.rlib,libcrc32fast-012a6ada2e528859.rlib,libsnap-f8176e9be181d92f.rlib,libfoyer-618524c67174d376.rlib,libfoyer_storage-0a07a61afebc50fd.rlib,liblz4-97ea552fc2cd216b.rlib,liblz4_sys-3562b61a95d22dd3.rlib,liblibc-1d740795677a515a.rlib,libanyhow-acd7bd43831136f1.rlib,libordered_hash_map-834dbfb88f10e43e.rlib,libhashbrown-b9b4a64d96d1a5c5.rlib,libahash-06717a81a20a2427.rlib,libzerocopy-4fc916ac2a54efcb.rlib,libzstd-88dfd21772d2e019.rlib,libzstd_safe-79b98a2fc9336856.rlib,libzstd_sys-873daef93bc74114.rlib,libflume-ae05636048cab8f6.rlib,libnanorand-c78cc8aea7b9b5cd.rlib,libspin-de863908697b1c86.rlib,libfs4-491f1a1ecad73950.rlib,libfoyer_memory-c1696e2eab603f38.rlib,libarc_swap-018394101cfaa8a8.rlib,libcmsketch-1633eb5133118d87.rlib,libfoyer_intrusive_collections-b14925ab9b7298ed.rlib,libmemoffset-f1221ea8c7fc2776.rlib,libfoyer_common-5518e2430e372e22.rlib,libbincode-5235fef40b7f77bb.rlib,libmixtrics-78e6c4e0a3c67f61.rlib,libpin_project-1abfda5929c37288.rlib,libtwox_hash-bbf56033878c5a50.rlib,libmoka-e711a96475abb717.rlib,libportable_atomic-f56b56173a34a6e0.rlib,libtagptr-e740f8703327172e.rlib,libasync_lock-fe59fb3e1c5ebda7.rlib,libevent_listener_strategy-a082d1f8b4d8aeab.rlib,libevent_listener-55b1c2530f95bda3.rlib,libparking-e21cf289b9587a33.rlib,libcrossbeam_channel-841b18811a9c0747.rlib,libdotenvy-c2d93914f750d463.rlib,libbackon-c64922c877043801.rlib,libfastrand-02e81ee75c4d97cd.rlib,libthread_local-92c33790d6b9d256.rlib,librand_xoshiro-abad714dcf8f0f16.rlib,libouroboros-25f23d68851cf7d9.rlib,libstatic_assertions-dc86ef657059a9cd.rlib,libaliasable-0d1372df8147ba71.rlib,libflatbuffers-99f590faccb7eb9d.rlib,libbitflags-9de86ad6eb4af16a.rlib,libsiphasher-fda3bf58e3c92d6d.rlib,libthiserror-86cc04ebe354eea8.rlib,libcrossbeam_skiplist-e8b5beccc91889b4.rlib,libcrossbeam_epoch-3b1f363218674468.rlib,libcrossbeam_utils-050512cf957ae886.rlib,libradix_trie-d19173ca566f663b.rlib,libnibble_vec-cbed3ffe68558b90.rlib,libendian_type-e9f05c6931aa4ea2.rlib,libatomic-2334222cf74113c0.rlib,libbytemuck-b8a7e0846554475e.rlib,libfigment-96ad047d736a7043.rlib,libserde_yaml-98ff3872ecd61998.rlib,libunsafe_libyaml-3d2e2eb50d51557e.rlib,libtoml-26a35c8a6c3d5d3d.rlib,libtoml_edit-4f0e08b032c8f86d.rlib,libserde_spanned-a9e32442f0d35a42.rlib,libwinnow-4c9bb9d1791c1e80.rlib,libtoml_datetime-19016a478cf47d73.rlib,libuncased-d317d9124111cf63.rlib,libpear-9cff97d7ee090da3.rlib,libyansi-c20cad4386ea27a6.rlib,libinlinable_string-e62d6da85719eb09.rlib,libduration_str-da3aab23edbc8a77.rlib,librust_decimal-aa77ae7f4c3db4a9.rlib,libarrayvec-bb2e20a36e5c84f5.rlib,libtime-2e21826429d2862b.rlib,libtime_core-532a5cdfcac6adde.rlib,libnum_conv-25ee515b4446d7e9.rlib,libderanged-6d12b6e72da3d584.rlib,libpowerfmt-a21659ed8be10e5c.rlib,libwinnow-d9bf16f024d15240.rlib,libulid-6b2f529948def3f5.rlib,libuuid-18907533ba3ff8cd.rlib,libobject_store-0d882ed0f9e8a477.rlib,libserde_json-e153d9a8857e94d8.rlib,libhumantime-fa8c0a33e6a89b65.rlib,libthiserror-85d369c7ef048559.rlib,libwalkdir-78d639485ab4e69a.rlib,libsame_file-998ece3fbac985ef.rlib,libwinapi_util-bfbabd022f47d1f4.rlib,libparking_lot-b71012324cd576a9.rlib,libparking_lot_core-3e777d769d49b5b0.rlib,liblock_api-f50801c8fd181135.rlib,libscopeguard-4a1404f096e49870.rlib,librand-5250d1b3d5aa6d11.rlib,librand_chacha-5973d736ed65a564.rlib,librand_core-478bcbd7b0a4446c.rlib,libgetrandom-e574ba26d9e78259.rlib,libchrono-1fc364279811952a.rlib,libnum_traits-362cf27f939c8b6b.rlib,libquick_xml-ae076ebb255afaff.rlib,libmd5-f072e389359dd181.rlib,libdigest-2ee16124d33029be.rlib,libblock_buffer-1ac3449d05088418.rlib,libcrypto_common-e63da4efaf5bda92.rlib,libgeneric_array-684a68f00a2647b4.rlib,libtypenum-f4e2056ab3ddd92c.rlib,libitertools-70d366f21c66e261.rlib,libeither-a42a57c60727e805.rlib,libreqwest-45ac7336a5e22fc6.rlib,libserde_urlencoded-ed2fe985f650ce49.rlib,libryu-6ac78a988ab9ad86.rlib,libbase64-c9cc1af3ab183922.rlib,librustls_pemfile-05af8cdd4ccb74df.rlib,libwindows_registry-5b79ae09d5fe2224.rlib,libwindows_targets-b590ae889d183991.rlib,libwindows_strings-086f3f1bd8aee5e6.rlib,libwindows_result-ef8eeaa26e13f8a1.rlib,libwindows_link-1906e3feb121b379.rlib,libipnet-7ce64db73d9ff15f.rlib,libhyper_rustls-c0950d4c78d8a1fe.rlib,librustls_native_certs-ddf88d38d15d8ee8.rlib,libschannel-84c169bbdbaf1a10.rlib,libtokio_rustls-b4de4f2a0a0852ca.rlib,librustls-884d928189f7f2a5.rlib,libsubtle-7a62713adcbc7e72.rlib,libwebpki-21511fda1dbcfb07.rlib,libring-2bd2abf3f3b39fc4.rlib,libuntrusted-1def58ec6b7b2f4e.rlib,libzeroize-c08345c222b1583f.rlib,librustls_pki_types-2e827a03ba10a040.rlib,libtower-e3ad644d9c1a564f.rlib,libsync_wrapper-e037a74fa6b7e2ad.rlib,libtower_layer-e54a78936e51308c.rlib,libhyper_util-631fae35aeac805f.rlib,libsocket2-93bc91d5cfb91dcb.rlib,libtower_service-85c893f18d7f8b9f.rlib,libhyper-1b0d839b9b16cec7.rlib,libwant-67e7bc29f1287585.rlib,libtry_lock-1d7644d8c8c15ffe.rlib,libhttparse-7a9cf2ec73ddb78e.rlib,libh2-2b912d297663f85a.rlib,libtracing-9a94ff0b859cca37.rlib,libtracing_core-0bc12b3517250cf2.rlib,libonce_cell-ecc86b59685e8d56.rlib,libindexmap-7e40a96b3f7029d3.rlib,libhashbrown-28b0f9102dac3ce4.rlib,libfoldhash-891f220795a7b7bc.rlib,libequivalent-0351353d24c6c3e5.rlib,liballocator_api2-54ceec082678ab87.rlib,libatomic_waker-f01c4339b1368ba1.rlib,libtokio_util-cf2aad0ab1bcc6c4.rlib,libtokio-4c7eaff8d2e3c478.rlib,libsocket2-3966143278ec7c36.rlib,libwindows_sys-ebd44454e4d04729.rlib,libmio-88bb0206e83ffecc.rlib,libwindows_sys-2ce4e0196f458452.rlib,libwindows_targets-dd8cca008988cf75.rlib,libhttp_body_util-eaca61c024567ef8.rlib,libhttp_body-dcf829490dfed289.rlib,liburl-bb3e18932487c83e.rlib,libidna-a345bb65df0d00cb.rlib,libidna_adapter-f1f7d1ac4935e46e.rlib,libicu_normalizer-a4624b93d661ade5.rlib,libicu_normalizer_data-666d8f3d46b55a2e.rlib,libwrite16-0f669802e9093f18.rlib,libutf8_iter-470e50d0e67000d5.rlib,libutf16_iter-27e504796cb8d100.rlib,libsmallvec-9581de2c2a3fa5ec.rlib,libicu_properties-046e1ed4d2ca22b4.rlib,libicu_properties_data-35b8ff2aee66a1d8.rlib,libicu_locid_transform-1b4ea2e54d26a370.rlib,libicu_locid_transform_data-213cc3c639c231fc.rlib,libicu_collections-37b3aefe9943bae5.rlib,libicu_provider-62092c2e0b3af566.rlib,libicu_locid-ddd06e2f031dff3d.rlib,liblitemap-34c4bc45f88d5ced.rlib,libtinystr-10e574e3c445f4d3.rlib,libzerovec-504b3c6f011bf39d.rlib,libwriteable-484bb341793a49d8.rlib,libyoke-1f4e9c5ff1ece3c5.rlib,libzerofrom-2ae76b6607d789ab.rlib,libstable_deref_trait-9c3667d1124708cb.rlib,libform_urlencoded-da3d6eb3563d5673.rlib,libpercent_encoding-36355d0f5ec6ec70.rlib,libhttp-02647beae3c50a3f.rlib,libitoa-04f76355b1fc7079.rlib,libfnv-98890e3ff67f2430.rlib,libfutures-d862adf37a767ea6.rlib,libfutures_executor-b0f6f46c937996d0.rlib,libfutures_util-991d311db73ffd1d.rlib,libmemchr-105646b5756cb14a.rlib,libfutures_io-5273fe5482d10a60.rlib,libslab-72f34d3e45a32922.rlib,libfutures_channel-fc933f5ba1427b0e.rlib,libpin_project_lite-efc1287e85ca2a83.rlib,libfutures_sink-aa7e954255f4c0c6.rlib,libfutures_task-1c3fa5431198fd55.rlib,libpin_utils-8ff768994ab1a462.rlib,libfutures_core-b17e2c2a87306059.rlib,libfail_parallel-3038093514060723.rlib,liblog-c603d8f956e435cc.rlib,librand-41caab4cc962eeeb.rlib,librand_chacha-40bc2e5eb375ed18.rlib,libppv_lite86-4635537977e6bada.rlib,libzerocopy-e74c5ede67148e91.rlib,librand_core-f4b1df4588c13156.rlib,libgetrandom-4038c272d2c4b354.rlib,libcfg_if-120b7212e7bc72ec.rlib,libbytes-0b03edaf76ca8073.rlib,libserde-7e9504e90e43872f.rlib}.rlib" "<sysroot>\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib/{libstd-*,libpanic_unwind-*,libwindows_targets-*,librustc_demangle-*,libstd_detect-*,libhashbrown-*,librustc_std_workspace_alloc-*,libunwind-*,libcfg_if-*,liballoc-*,librustc_std_workspace_core-*,libcore-*,libcompiler_builtins-*}.rlib" "C:\\Users\\rebond\\AppData\\Roaming\\uv\\python\\cpython-3.10.15-windows-x86_64-none\\libs\\python310.lib" "legacy_stdio_definitions.lib" "C:\\Users\\rebond\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib\\windows.0.52.0.lib" "C:\\Users\\rebond\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.53.0\\lib\\windows.0.53.0.lib" "C:\\Users\\rebond\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib\\windows.0.52.0.lib" "C:\\Users\\rebond\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib\\windows.0.52.0.lib" "bcrypt.lib" "advapi32.lib" "kernel32.lib" "kernel32.lib" "ntdll.lib" "userenv.lib" "ws2_32.lib" "dbghelp.lib" "/defaultlib:msvcrt" "/NXCOMPAT" "/LIBPATH:C:\\Users\\rebond\\AppData\\Roaming\\uv\\python\\cpython-3.10.15-windows-x86_64-none\\libs" "/LIBPATH:C:\\Users\\rebond\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib" "/LIBPATH:C:\\Program Files\\Microsoft Visual Studio\\18\\Preview\\VC\\Tools\\MSVC\\14.50.35503\\atlmfc\\lib\\x64" "/LIBPATH:C:\\dev\\slatedb\\target\\debug\\build\\lz4-sys-74d40dd2b78c4c74\\out" "/LIBPATH:C:\\Program Files\\Microsoft Visual Studio\\18\\Preview\\VC\\Tools\\MSVC\\14.50.35503\\atlmfc\\lib\\x64" "/LIBPATH:C:\\dev\\slatedb\\target\\debug\\build\\zstd-sys-3018e183507f2198\\out" "/LIBPATH:C:\\dev\\slatedb\\target\\debug\\build\\ring-1dcc49f4e9fdae08\\out" "/LIBPATH:C:\\Users\\rebond\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.53.0\\lib" "/OUT:C:\\dev\\slatedb\\target\\debug\\deps\\slatedb.dll" "/OPT:REF,NOICF" "/DLL" "/IMPLIB:C:\\dev\\slatedb\\target\\debug\\deps\\slatedb.dll.lib" "/DEBUG" "/PDBALTPATH:%_PDB%" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\libstd.natvis"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: LINK : fatal error LNK1201: error writing to program database 'C:\dev\slatedb\target\debug\deps\slatedb.pdb'; check for insufficient disk space, invalid path, or insufficient privilege␍


error: could not compile `slatedb-py` (lib) due to 1 previous error
```

</details>

**To Reproduce**

Run `cargo build` from repo root on Windows.

**Expected behavior**

It builds without error.

--- a/slatedb-py/Cargo.toml
+++ b/slatedb-py/Cargo.toml
@@ -9,7 +9,7 @@ license.workspace = true
 homepage.workspace = true
 
 [lib]
-name = "slatedb"
+name = "slatedb_py"
 crate-type = ["cdylib"]
 doc = false
 # Disable `libtest` harness because it fights with Criterion's `--output-format bencher`

